vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -199,47 +200,73 @@ public Retrofit getRetrofit() {
+     */
     public void setVimeoAccount(@Nullable VimeoAccount vimeoAccount) {
         if (vimeoAccount == null) {
-            // If the provided account was null but we have an access token, persist the vimeo account with
-            // just a token in it. Otherwise we'll want to leave the persisted account as null.
-            vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
             if (mConfiguration.accessToken != null) {
-                mConfiguration.saveAccount(vimeoAccount, null);
+                // If the provided account was null but we have an access token, persist the vimeo account with
+                // just a token in it. Otherwise we'll want to leave the persisted account as null.
+                vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
+                mConfiguration.saveNonUserAccount(vimeoAccount);
             }
         }
 
         mVimeoAccount = vimeoAccount;
     }
 
     /**
-     * @deprecated use {@link #saveAccount(VimeoAccount, String)} instead
-     * <p/>
-     * We find no use in storing the password when you can persist the {@link VimeoAccount} across
-     * application sessions.
+     * Sets the {@link #mVimeoAccount} field as well as triggering the saveAuthenticatedUserAccount
+     * event for the account store.
+     *
+     * @param vimeoAccount The account to save - this should be provided; if it is not, then one will
+     *                     be constructed using the access token set on the {@link Configuration}.
+     * @param accountName  accountName should be provided if the account is for an authenticated user. It may
+     *                     be null if the account represents a client credentials grant.
      */
-    @Deprecated
-    public void saveAccount(@Nullable VimeoAccount vimeoAccount, String email, String password) {
-        saveAccount(vimeoAccount, email);
+    public void saveAuthenticatedUserAccount(@Nullable VimeoAccount vimeoAccount,
+                                             @NotNull String accountName) {
+        setVimeoAccount(vimeoAccount);
+        if (vimeoAccount != null) {
+            mConfiguration.saveAuthenticatedUserAccount(vimeoAccount, accountName);
+        }
     }
 
     /**
-     * Sets the {@link #mVimeoAccount} field as well as triggering the saveAccount event for the
-     * account store
+     * Sets the {@link #mVimeoAccount} field as well as triggering the saveNonUserAccount event for the
+     * account store. This is meant to be user when no user is directly associated with the account.
+     *
+     * @param vimeoAccount The account to save - this should be provided; if it is not, then one will
+     *                     be constructed using the access token set on the {@link Configuration}.
      */
-    public void saveAccount(@Nullable VimeoAccount vimeoAccount, String email) {
+    public void saveNonUserAccount(@Nullable VimeoAccount vimeoAccount) {
         setVimeoAccount(vimeoAccount);
```

Comment:
Seems like we're relying on similar logic here. In this case with `setVimeoAccount` doing some opaque stuff in response to a null account object.